rdf_differ/skos_history_wrapper.py

```diff
@@ -32,37 +32,43 @@
 INPUT_MIME_TYPE=\"{input_type}\""""
 
 
+class SKOSException(Exception):
+    """
+        An exception for SKOSHistoryRunner.
+    """
+
+
 class SKOSHistoryRunner:
-    def __init__(self, dataset: str, scheme_uri: str, basedir: str, filename: str, endpoint: str,
-                 old_version_file: str, new_version_file: str, old_version_id: str, new_version_id: str,
+    def __init__(self, dataset: str, scheme_uri: str, old_version_file: str, new_version_file: str, old_version_id: str,
+                 new_version_id: str, basedir: str, filename: str = None, endpoint: str = None,
                  config_template: str = CONFIG_TEMPLATE):
         """
         Class for running the skos-history shell script.
         It includes folder structure creation and config file population.
 
         :param dataset: the name used
         :param scheme_uri: the concept scheme or dataset URI
-        :param basedir: location for folder generation
-        :param filename: the name of the file to be used for upload
-        (its extension will not be taken into consideration if given)
-        :param endpoint: upload url
         :param old_version_file: the location of the file to be uploaded
         :param new_version_file: the location of the file to be uploaded
         :param old_version_id: name used for diff upload
         :param new_version_id: name used for diff upload
+        :param basedir: location for folder generation
+        :param filename: the name of the file to be used for upload
+        (its extension will not be taken into consideration if given)
+        :param endpoint: upload url
         :param config_template: string
 
         file_format: format of the files used, as defined in INPUT_MIME_TYPES
         file_extension: extension of the files used, as defined in INPUT_MIME_TYPES
         """
         if not (dataset and scheme_uri and old_version_file and old_version_id and new_version_file and new_version_id):
-            raise Exception('These parameters cannot be empty:'
-                            f'{" dataset" if not dataset else ""}'
-                            f'{" scheme_uri" if not scheme_uri else ""}'
-                            f'{" old_version_file" if not old_version_file else ""}'
-                            f'{" old_version_id" if not old_version_id else ""}'
-                            f'{" new_version_file" if not new_version_file else ""}'
-                            f'{" new_version_id." if not new_version_id else "."}')
+            raise SKOSException('These parameters cannot be empty:'
```

Comment:
you may want to check the parameters separately and 
`raise ValueError('This parameter is mandatory')` where needed

Reply:
Or AttributeError